File: InvestmentPortfolioTracker/src/InvestmentPortfolioTracker.Core/Models/AccountAggregate/Enums/AccountType.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace InvestmentPortfolioTracker.Core;

/// <summary>
/// Represents the type of investment account.
/// </summary>
public enum AccountType
{
    /// <summary>
    /// Standard taxable brokerage account.
    /// </summary>
    Taxable = 0,

    /// <summary>
    /// Traditional IRA.
    /// </summary>
    TraditionalIRA = 1,

    /// <summary>
    /// Roth IRA.
    /// </summary>
    RothIRA = 2,

    /// <summary>
    /// 401(k) retirement account.
    /// </summary>
    FourZeroOneK = 3,

    /// <summary>
    /// 403(b) retirement account.
    /// </summary>
    FourZeroThreeB = 4,

    /// <summary>
    /// Health Savings Account (HSA).
    /// </summary>
    HSA = 5,

    /// <summary>
    /// 529 college savings plan.
    /// </summary>
    FiveTwoNine = 6,
}

File: InvestmentPortfolioTracker/src/InvestmentPortfolioTracker.Web/projects/api/src/lib/models/account-type.ts
export enum AccountType {
  Taxable = 0,
  TraditionalIRA = 1,
  RothIRA = 2,
  FourZeroOneK = 3,
  FourZeroThreeB = 4,
  HSA = 5,
  FiveTwoNine = 6
}

export const ACCOUNT_TYPE_LABELS: Record<AccountType, string> = {
  [AccountType.Taxable]: 'Taxable',
  [AccountType.TraditionalIRA]: 'Traditional IRA',
  [AccountType.RothIRA]: 'Roth IRA',
  [AccountType.FourZeroOneK]: '401(k)',
  [AccountType.FourZeroThreeB]: '403(b)',
  [AccountType.HSA]: 'HSA',
  [AccountType.FiveTwoNine]: '529'
};
